freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/model/UsersStateDifference.java

```diff
@@ -113,9 +125,42 @@ public static UsersStateDifference forDeletedUser(String deletedUser, Collection
         return usersToAdd;
     }
 
-    public static ImmutableSet<String> calculateUsersToRemove(UmsUsersState umsState, UsersState ipaState) {
+    public static ImmutableSet<String> calculateUsersWithCredentialsToUpdate(UmsUsersState umsState, FreeIpaUsersState ipaState,
```

Comment:
This whole class still assumes that, if the FreeIPA username matches, then it's the same user in the control plane. Do we have another JIRA open for using the crn for these matches instead?

Reply:
Right, in this change I didn't want to introduce any functional differences not related to the password hash optimization -- beyond storing the CRN, which it makes sense to do now to avoid backward compatibility headaches later. 

CDPCP-892 is related, but not exactly what you're talking about. I think there may be a few other related JIRAs floating around. I'll take a look, and file a new one if there isn't an appropriate one already.